src/NHibernate/Hql/Ast/HqlTreeBuilder.cs

```diff
@@ -363,6 +363,11 @@ public HqlJoin Join(HqlExpression expression, HqlAlias @alias)
 			return new HqlJoin(_factory, expression, @alias);
 		}
 
+		public HqlInnerJoin InnerJoin(HqlExpression expression, HqlAlias @alias)
+		{
+			return new HqlInnerJoin(_factory, expression, @alias);
```

Comment:
Unfortunately new join feature does not support implicit join types, so have to be explicit.